pinot-broker/src/main/java/org/apache/pinot/broker/broker/AuthenticationFilter.java

```diff
@@ -81,10 +83,32 @@ public void filter(ContainerRequestContext requestContext)
 
     HttpRequesterIdentity httpRequestIdentity = HttpRequesterIdentity.fromRequest(request);
 
+    // default authorization handling
     if (!accessControl.hasAccess(httpRequestIdentity)) {
       throw new WebApplicationException("Failed access check for " + httpRequestIdentity.getEndpointUrl(),
           Response.Status.FORBIDDEN);
     }
+
+    handleRBACAuthorization(endpointMethod, uriInfo, accessControl, httpRequestIdentity);
+  }
+
+  private void handleRBACAuthorization(Method endpointMethod, UriInfo uriInfo, AccessControl accessControl,
+                                       HttpRequesterIdentity httpRequestIdentity) {
+    if (endpointMethod.isAnnotationPresent(RBACAuthorization.class)) {
+      RBACAuthorization rbacAuthorization = endpointMethod.getAnnotation(RBACAuthorization.class);
+      String targetId = RBACAuthUtils.getTargetId(rbacAuthorization.targetId(), uriInfo.getPathParameters(),
```

Comment:
I feel that we should use different names for the variable `targetId` and `rbacAuthorization.targetId()`, it is confusing to me now.